### PR TITLE
project_key: fix controller tests

### PR DESCRIPTION
**Pull request by simahawk**
blind change, not tested, let's see if it fixes https://github.com/OCA/project/pull/846

### CODE REVIEW

**Comment by simahawk** [comment]
~~BTW this fix is not even 100% correct because we should call the endpoint w/ `url_open` instead.~~

File: project_key/tests/test_controller.py [hunk 1]
```diff
@@ -1,32 +1,16 @@
...
     def test_02_task_browse(self):
-        with patch.object(http, "request") as request:
-            request.env = self.env
-            controller = ProjectBrowser()
-            response = controller.open_task(self.task11.key)
-
-            self.assertEqual(response.status_code, 301)
-            self.assertEqual(response.location, self.get_task_url(self.task11))
+        response = self.url_open("/tasks/" + self.task11.key)
+        self.assertEqual(response.status_code, 301)
```
Comment (AntoniRomera) [inline project_key/tests/test_controller.py]:
```suggestion
        self.assertEqual(response.status_code, 200)
```

File: project_key/tests/test_controller.py [hunk 2]
```diff
@@ -1,32 +1,16 @@
...
 
     def test_02_task_browse(self):
-        with patch.object(http, "request") as request:
-            request.env = self.env
-            controller = ProjectBrowser()
-            response = controller.open_task(self.task11.key)
-
-            self.assertEqual(response.status_code, 301)
-            self.assertEqual(response.location, self.get_task_url(self.task11))
+        response = self.url_open("/tasks/" + self.task11.key)
```
Comment (AntoniRomera) [inline project_key/tests/test_controller.py:18]:
```suggestion
        self.authenticate("demo", "demo")
        response = self.url_open("/tasks/" + self.task11.key)
```

File: project_key/tests/test_controller.py [hunk 3]
```diff
@@ -1,32 +1,16 @@
...
     def test_01_project_browse(self):
-        with patch.object(http, "request") as request:
-            request.env = self.env
-            controller = ProjectBrowser()
-            response = controller.open_project(self.project_1.key)
-
-            self.assertEqual(response.status_code, 301)
-            self.assertEqual(response.location, self.get_project_url(self.project_1))
+        response = self.url_open("/projects/" + self.project_1.key)
+        self.assertEqual(response.status_code, 301)
```
Comment (AntoniRomera) [inline project_key/tests/test_controller.py]:
```suggestion
        self.assertEqual(response.status_code, 200)
```

File: project_key/tests/test_controller.py [hunk 4]
```diff
@@ -1,32 +1,16 @@
...
-
     def test_01_project_browse(self):
-        with patch.object(http, "request") as request:
-            request.env = self.env
-            controller = ProjectBrowser()
-            response = controller.open_project(self.project_1.key)
-
-            self.assertEqual(response.status_code, 301)
-            self.assertEqual(response.location, self.get_project_url(self.project_1))
+        response = self.url_open("/projects/" + self.project_1.key)
```
Comment (AntoniRomera) [inline project_key/tests/test_controller.py:10]:
```suggestion
        self.authenticate("demo", "demo")
        response = self.url_open("/projects/" + self.project_1.key)
```

File: project_key/tests/test_controller.py [hunk 5]
```diff
@@ -1,32 +1,16 @@
...
+        self.assertEqual(response.status_code, 301)
+        self.assertEqual(response.location, self.get_project_url(self.project_1))
```
Comment (AntoniRomera) [inline project_key/tests/test_controller.py]:
```suggestion
        self.assertEqual(("/{}".format(response.url.split('/')[-1]),, self.get_project_url(self.project_1))
```

File: project_key/tests/test_controller.py [hunk 6]
```diff
@@ -1,32 +1,16 @@
...
-        with patch.object(http, "request") as request:
-            request.env = self.env
-            controller = ProjectBrowser()
-            response = controller.open_task(self.task11.key)
-
-            self.assertEqual(response.status_code, 301)
-            self.assertEqual(response.location, self.get_task_url(self.task11))
+        response = self.url_open("/tasks/" + self.task11.key)
+        self.assertEqual(response.status_code, 301)
+        self.assertEqual(response.location, self.get_task_url(self.task11))
```
Comment (AntoniRomera) [inline project_key/tests/test_controller.py]:
```suggestion
        self.assertEqual(("/{}".format(response.url.split('/')[-1]),, self.get_task_url(self.task11))
```

**Comment by AntoniRomera** [comment]
Thanks for the help, I'll give you this review with the things I've seen and the knowledge I have.

**Comment by simahawk** [comment]
yeah, sorry I did it too fast (TBH I don't have time now :sweat_smile: )
I fixed it and did another change to boost test speed as the setup was quite expensive (running for each and every test).

**Comment by simahawk** [comment]
@AntoniRomera @pedrobaeza good for you?

**Review by AntoniRomera — APPROVED** [review]
I think it's right, thank you very much for doing it on short notice.

**Comment by pedrobaeza** [comment]
Thanks for doing it so quickly and interrupting your work!

/ocabot merge patch

**Pull request merged**